operate/importer-8_4/src/main/java/io/camunda/operate/zeebeimport/v8_4/processors/FlowNodeInstanceZeebeRecordProcessor.java

```diff
@@ -74,6 +93,7 @@ public FlowNodeInstanceZeebeRecordProcessor(
             flowNodeStore::findParentTreePathFor,
             new TreePathCacheMetricsImpl(partitionIds, metrics));
     fniTransformer = new FNITransformer(treePathCache);
+    return fniTransformer;
```

Comment:
Should we synchronize the creation of the transformer? Otherwise there is a race condition that we create more than one, although I don't see an immediate negative side effect of it. But is a future bug potential.